transport/tchannel/transport.go

```diff
@@ -201,10 +229,24 @@ func (t *Transport) Stop() error {
 
 func (t *Transport) stop() error {
 	t.ch.Close()
+	t.connectorsGroup.Wait()
```

Comment:
if one of the subscribers doesn't properly exit, this wait will remain forever.  Should we have a "purge" of the peer list before we wait?  (Or at least validate that we're not retaining any peers)